codegen/post_gen_hooks.go

```diff
@@ -23,114 +23,229 @@ package codegen
-func ClientMockGenHook(h *PackageHelper) (PostGenHook, error) {
-	bin, err := NewMockgenBin(h.PackageRoot())
+func ClientMockGenHook(h *PackageHelper, t *Template) (PostGenHook, error) {
+	bin, err := NewMockgenBin(h, t)
 	if err != nil {
 		return nil, errors.Wrap(err, "error building mockgen binary")
 	}
 
 	return func(instances map[string][]*ModuleInstance) error {
 		fmt.Println("Generating client mocks:")
 		mockCount := len(instances["client"])
+		ec := make(chan error, mockCount)
+		var files sync.Map
+		var wg sync.WaitGroup
+		wg.Add(mockCount)
 		for i, instance := range instances["client"] {
-			if err := bin.GenMock(
-				instance,
-				"mock-client/mock_client.go",
-				"clientmock",
-				"Client",
-			); err != nil {
-				return errors.Wrapf(
-					err,
-					"error generating mocks for client %q",
+			go func(instance *ModuleInstance, i int) {
+				defer wg.Done()
+
+				var mc moduleConfig
+				if err := json.Unmarshal(instance.JSONFileRaw, &mc); err != nil {
+					ec <- errors.Wrapf(
+						err,
+						"error parsing client-config.json for client %q",
+						instance.InstanceName,
+					)
+					return
+				}
+
+				buildDir := h.CodeGenTargetPath()
+				genDir := filepath.Join(buildDir, instance.Directory, "mock-client")
+
+				importPath := instance.PackageInfo.GeneratedPackagePath
+				if instance.ClassType == "custom" {
+					importPath = mc.Config.CustomImportPath
+					if importPath == "" {
+						ec <- errors.Errorf("custom client %q must have customImportPath", instance.ClassName)
+						return
+					}
+				}
+
+				// generate mock client
+				mock, err := bin.GenMock(importPath, "clientmock", clientInterface)
+				if err != nil {
+					ec <- errors.Wrapf(
+						err,
+						"error generating mocks for client %q",
+						instance.InstanceName,
+					)
+					return
+				}
+				files.Store(filepath.Join(genDir, "mock_client.go"), mock)
+
+				// generate fixture types and augmented mock client
+				f := mc.Config.Fixture
+				if f != nil && f.Scenarios != nil {
+					types, augMock, err := bin.AugmentMockWithFixture(importPath, f, clientInterface)
+					if err != nil {
+						ec <- errors.Wrapf(
+							err,
+							"error generating mock client with fixtures for client %q",
+							instance.InstanceName,
+						)
+						return
+					}
+
+					files.Store(filepath.Join(genDir, "types.go"), types)
+					files.Store(filepath.Join(genDir, "mock_client_with_fixture.go"), augMock)
+				}
+
+				fmt.Printf(
+					genFormattor,
+					"mock",
+					instance.ClassName,
 					instance.InstanceName,
+					path.Join(path.Base(buildDir), instance.Directory, "mock-client"),
+					i+1, mockCount,
 				)
-			}
-			fmt.Printf(
-				"Generating %12s %12s %-30s in %-50s %d/%d\n",
-				"mock",
-				instance.ClassName,
-				instance.InstanceName,
-				path.Join(path.Base(h.CodeGenTargetPath()), instance.Directory, "mock-client"),
-				i+1, mockCount,
-			)
+			}(instance, i)
+		}
+		wg.Wait()
+
+		select {
```

Comment:
If there is an issue with multiple mocks, then I would need to run this repeatedly to figure out these errors. If you are going to run all of them unconditionally, then you should show the user the errors from all of them. Collect the errors and return a single error that accumulates all of the errors, i.e. `err = errors.New("Encountered %d errors when generating mocks: ", N, strings.Join("\n", err1, err2, ..., errN))`